File: packages/core/src/core/content-signal.ts
import type { ContentSignalOptions } from '../types.js';

/**
 * Build the value for a `content-signal` HTTP header from the given options.
 *
 * Only includes directives that are explicitly set (not `undefined`).
 * Returns `undefined` when no directives are set.
 *
 * @example
 * ```ts
 * buildContentSignalHeader({ aiTrain: true, search: true, aiInput: true });
 * // → "ai-train=yes, search=yes, ai-input=yes"
 * ```
 */
export function buildContentSignalHeader(options: ContentSignalOptions): string | undefined {
    const parts: string[] = [];

    if (options.aiTrain !== undefined) {
        parts.push(`ai-train=${options.aiTrain ? 'yes' : 'no'}`);
    }
    if (options.search !== undefined) {
        parts.push(`search=${options.search ? 'yes' : 'no'}`);
    }
    if (options.aiInput !== undefined) {
        parts.push(`ai-input=${options.aiInput ? 'yes' : 'no'}`);
    }

    return parts.length > 0 ? parts.join(', ') : undefined;
}

File: python/src/markdown_for_agents/core/content_signal.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .._types import ContentSignalOptions


def build_content_signal_header(options: ContentSignalOptions) -> str | None:
    """Build the value for a content-signal HTTP header from the given options."""
    parts: list[str] = []

    if options.ai_train is not None:
        parts.append(f"ai-train={'yes' if options.ai_train else 'no'}")
    if options.search is not None:
        parts.append(f"search={'yes' if options.search else 'no'}")
    if options.ai_input is not None:
        parts.append(f"ai-input={'yes' if options.ai_input else 'no'}")

    return ", ".join(parts) if parts else None
